packages/studio-ui/src/web/components/Layout/StatusBar/LangSwitcher.tsx

```diff
@@ -1,122 +1,65 @@
-import { lang, utils } from 'botpress/shared'
+import { Button } from '@blueprintjs/core'
+import { Dropdown } from 'botpress/shared'
 import _ from 'lodash'
-import React, { Fragment } from 'react'
-import { Dropdown } from 'react-bootstrap'
+import React, { useEffect } from 'react'
 
 import withLanguage from '../../Util/withLanguage'
 
-import ActionItem from './ActionItem'
-import style from './style.scss'
-
-const requireFlag = (code) => {
-  try {
-    return require(`../../../img/flags/${code}.svg`)
-  } catch (err) {
-    return requireFlag('missing')
-  }
-}
+import { Flag } from './Flag'
 
 interface Props {
   contentLang: string
-  languages: any
+  languages: string[]
   changeContentLanguage: any
   toggleLangSwitcher: any
   langSwitcherOpen: boolean
 }
 
 const STORAGE_KEY = `bp::${window.BOT_ID}::cmsLanguage`
 
-class LangSwitcher extends React.Component<Props> {
-  elems = {}
-
-  componentDidMount() {
-    this.restoreLastLanguage()
-  }
+export const LanguageSwitcher = (props: Props) => {
+  useEffect(() => {
+    restoreLastLanguage()
```

Comment:
I don't think you need the `restoreLastLanguage` function– you can just move the contents of the function inside of this useEffect

Reply:
done ✅